networkx/algorithms/community/demon.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+#    Copyright Giulio Rossetti, Michele Coscia (C) 2015
+#    All rights reserved.
+#    BSD license.
+
+"""
+Flat Merge version of DEMON (Democratic Estimate of the Modular Organization of
+a Network) algorithm. DEMON uses a local-first approach to community discovery 
+were each node democratically votes for the communities that it sees in its
+surroundings. It provides a deterministic, fully incremental method for 
+community discovery that has a limited time complexity.
+"""
+
+from __future__ import division
+import networkx as nx
+import random
+from collections import defaultdict
+
+
+__all__ = ["demon_communities"]
+
+
+def demon_communities(G, epsilon=0.25, min_community_size=3, weight=None):
+    """Finds the communities in a graph using the label Democratic Estimate of
+    the Modular Organization of a Network (DEMON) algorithm[1]_. This method
+    uses a local-first approach to community discovery were each node
+    democratically votes for the communities that it sees in its surroundings.
+    Communities in this algorithm can be overlapping.
+
+    Parameters
+    ----------
+    G : graph
+        A NetworkX graph.
+
+    epsilon: real
+        The tolerance required in order to merge communities. The epsilon
+        factor is introduced to vary the percentage of common elements provided
+        from each couple of communities: epsilon = 0 ensure that two
+        communities are merged only if one of them is a proper subset of the
+        other, on the other hand with a value of epsilon = 1 even communities
+        that do not share a single node are merged together
+
+    min_community_size: integer
+        Min nodes needed to form a community
+
+    weight: String
+        If the graph is weighted, the edge parameter representing the weight
+
+    Returns
+    -------
+    communities : list
+        List of overlapping communities (sets)
+
+    References
+    ----------
+    .. [1] Michele Coscia, Giulio Rossetti, Fosca Giannotti, Dino Pedreschi:
+           DEMON: a local-first discovery method for overlapping communities.
+           KDD 2012:615-623.
+    """
+
+    all_communities = []
+
+    for ego in G:
+        # EgoMinusEgo and LabelPropagation phase
+        ego_minus_ego = nx.ego_graph(G, ego, 1, False)
+        community_to_nodes = _overlapping_label_propagation(ego_minus_ego, ego, weight)
+
+        # merging phase
+        for c in community_to_nodes.keys():
+            if len(community_to_nodes[c]) > min_community_size:
+                actual_community = community_to_nodes[c]
+                all_communities = _merge_communities(all_communities, set(actual_community), epsilon)
+
+    for c in all_communities:
+        yield c
+
+
+def _overlapping_label_propagation(ego_minus_ego, ego, weight, max_iter=100):
```

Comment:
Is this just the label propagation algorithm from pull request #1559 (or maybe #1259)? If so, just use that function.